src/plugins/intel_cpu/CMakeLists.txt

```diff
@@ -21,18 +21,33 @@ elseif(OV_COMPILER_IS_CLANG)
     ov_add_compiler_flags(-Wno-delete-non-abstract-non-virtual-dtor)
 endif()
 
+set(OV_CPU_ARM_TARGET_GENERIC_ARCHS armv8a
+                                    armv8.2-a
+                                    armv8.6-a armv8.6-a-sve armv8.6-a-sve2 armv8.6-a-sve2-sme2
+                                    armv8r64 # the same as armv8.4-a
+)
 if(ARM)
     set(OV_CPU_ARM_TARGET_ARCH_DEFAULT armv7a)
+    set(OV_CPU_ARM_TARGET_ARCHS armv7a armv7a-hf
+                                # requires estate=32
+                                ${OV_CPU_ARM_TARGET_GENERIC_ARCHS})
 elseif(AARCH64)
     if(APPLE)
         # Apple M1 / M2 is assumed
         set(OV_CPU_ARM_TARGET_ARCH_DEFAULT arm64-v8.2-a)
-        add_compile_definitions(OV_CPU_ARM_ENABLE_FP16)
     else()
         set(OV_CPU_ARM_TARGET_ARCH_DEFAULT arm64-v8a)
     endif()
+    set(OV_CPU_ARM_TARGET_ARCHS arm64-v8a
+                                arm64-v8.2-a arm64-v8.2-a-sve arm64-v8.2-a-sve2
+                                # used with estate=64
+                                ${OV_CPU_ARM_TARGET_GENERIC_ARCHS})
+endif()
+set(OV_CPU_ARM_TARGET_ARCH ${OV_CPU_ARM_TARGET_ARCH_DEFAULT} CACHE STRING "Architecture for ARM ComputeLibrary")
+set_property(CACHE OV_CPU_ARM_TARGET_ARCH PROPERTY STRINGS ${OV_CPU_ARM_TARGET_ARCHS})
+if(OV_CPU_ARM_TARGET_ARCH MATCHES "arm64-v(8|9).[0-9]-a" AND NOT OV_CPU_ARM_TARGET_ARCH MATCHES "arm64-v8.(0|1)-a")
```

Comment:
`.` in `MATCHES` statement means "any symbol".
I suppose we need to use something like:
```
MATCHES "(armv|arm64-v)[8-9]\\."
```

Reply:
suggested regexp works well, updated PR